### ISSUE TITLE
Update the sha1 crate to 0.10

### ISSUE THREAD

**Issue by sylvestre** [uutils/coreutils#2939]
We are using 0.6 currently:
```

Cargo.toml:sha1 = { version="0.6", features=["std"] }
src/uu/hashsum/Cargo.toml:sha1 = "0.6.0"
```

https://deps.rs/crate/sha1/0.6.1

**Comment by Aman-Mandal** [comment]
Hey.. Can I work on this ? And if Yes, can you guide .. am very new to open source and this is my first issue

**Comment by sylvestre** [comment]
sure, please go ahead :)

**Comment by Aman-Mandal** [comment]
But I don't know anything about it! Can anybody help me out? :/

**Comment by sylvestre** [comment]
well, discovering yourself is part of the exercise ;)

**Comment by Aman-Mandal** [comment]
Gotcha :)

**Comment by tertsdiepraam** [comment]
If you're really starting from zero, I recommend this guide on open source development: https://opensource.guide/how-to-contribute/ (specifically the part on [pull requests](https://opensource.guide/how-to-contribute/#opening-a-pull-request)). But since you already have some repositories of your own, you're probably already familiar with all of that.

Once you create a PR, the automated CI will start running and check:
- spelling
- linting (via clippy)
- formatting (via rustfmt)
- whether it builds for all platforms
- whether the tests pass for all platforms
- test against the GNU tests
- code coverage

Don't worry if it the CI does not immediately pass on the first try. That's what it's there for. 😄 But do try to run the tests, formatting and clippy before making the PR locally.

We will then review the PR and either accept it or request some changes. If you need to change some things, you can just push commits to the branch you made the PR for and the CI will rerun.

As for this specific issue, I agree with Sylvestre ;)

**Comment by Aman-Mandal** [comment]
Hey..I submitted a PR. Can you check please ? Thankyouu :))

**Comment by ybc37** [comment]
I guess this issue should be closed as #3110 was merged :checkered_flag:
